perfkitbenchmarker/providers/openstack/os_disk.py

```diff
@@ -24,96 +24,198 @@
-
-        is_unavailable = True
-        while is_unavailable:
-            time.sleep(1)
-            volume = self.__nclient.volumes.get(self._disk.id)
-            if volume:
-                is_unavailable = not (volume.status == "available")
-                self._disk = volume
-
-    @retry_authorization(max_retries=4)
-    def _Delete(self):
-        from novaclient.exceptions import NotFound
-
-        if self._disk is None:
-            logging.info('Volume %s was not created. Skipping deletion.'
-                         % self.name)
-            return
-
-        sleep = 1
-        sleep_count = 0
-        try:
-            self.__nclient.volumes.delete(self._disk)
-            is_deleted = False
-            while not is_deleted:
-                volume = self.__nclient.volumes.get(self._disk.id)
-                is_deleted = volume is None
-                time.sleep(sleep)
-                sleep_count += 1
-                if sleep_count == 10:
-                    sleep = 5
-        except NotFound:
-            logging.info('Volume %s not found, might have been already deleted'
-                         % self._disk.id)
-
-    def _Exists(self):
-        from novaclient.exceptions import NotFound
-        try:
-            volume = self.__nclient.volumes.get(self._disk.id)
-            return volume and volume.status in ('available', 'in-use',
-                                                'deleting',)
-        except NotFound:
-            return False
-
-    def Attach(self, vm):
-        self.attached_vm_name = vm.name
-        self.attached_vm_id = vm.id
-
-        result = self.__nclient.volumes.create_server_volume(vm.id,
-                                                             self._disk.id,)
-        self.attach_id = result.id
-
-        volume = None
-        is_unattached = True
-        while is_unattached:
-            time.sleep(1)
-            volume = self.__nclient.volumes.get(result.id)
-            if volume:
-                is_unattached = not(volume.status == "in-use"
-                                    and volume.attachments)
-
-        for attachment in volume.attachments:
-            if self.attach_id == attachment.get('volume_id'):
-                self.device = attachment.get('device')
-                return
-
-        raise errors.Error("Couldn't not attach volume to %s" % vm.name)
-
-    def GetDevicePath(self):
-        return self.device
-
-    def Detach(self):
-        self.__nclient.volumes.delete_server_volume(self.attached_vm_id,
-                                                    self.attach_id)
+
+class OpenStackDisk(disk.BaseDisk):
+  """
+  Base class for OpenStack disks.
+  """
+  def __init__(self, disk_spec, name):
+    super(OpenStackDisk, self).__init__(disk_spec)
+    self.name = name
+
+  def _Create(self):
+    pass
+
+  def _Delete(self):
+    pass
+
+  def Attach(self, vm):
+    pass
+
+  def Detach(self):
+    pass
+
+
+class OpenStackBlockStorageDisk(OpenStackDisk):
```

Comment:
Consider naming this `OpenStackRemoteBlockStorageDisk` instead, because otherwise it sounds like it could represent a local drive as well as a remote one.